src/concrete/ml/sklearn/tree.py

```diff
@@ -234,7 +236,9 @@ def load_dict(cls, metadata: Dict):
         obj._is_compiled = metadata["_is_compiled"]
         obj.input_quantizers = metadata["input_quantizers"]
         obj.framework = metadata["framework"]
-        obj._tree_inference, obj.output_quantizers, obj.onnx_model_ = tree_to_numpy(
+        obj.onnx_model_ = metadata["onnx_model_"]
+        obj.output_quantizers = metadata["output_quantizers"]
+        obj._tree_inference, _, _ = tree_to_numpy(
```

Comment:
You are recomputing the tree_inference without the use_rounding here so the inference will be difference than the saved model.